airflow/contrib/hooks/spark_submit_hook.py

```diff
@@ -174,8 +174,7 @@ def _resolve_should_track_driver_status(self):
         subsequent spark-submit status requests after the initial spark-submit request
         :return: if the driver status should be tracked
         """
-        return ('spark://' in self._connection['master'] and
-                self._connection['deploy_mode'] == 'cluster')
+        return self._connection['deploy_mode'] == 'cluster'
```

Comment:
return (('spark://' in self._connection['master'] or conn_type starts with spark) and self._connection['deploy_mode'] == 'cluster')